bin/ch_core.c

```diff
@@ -46,11 +46,29 @@ struct bind BINDS_DEFAULT[] = {
    { "/etc/hosts",               "/etc/hosts",               BD_OPTIONAL },
    { "/etc/machine-id",          "/etc/machine-id",          BD_OPTIONAL },
    { "/etc/resolv.conf",         "/etc/resolv.conf",         BD_OPTIONAL },
-   { "/var/lib/hugetlbfs",       "/var/opt/cray/hugetlbfs",  BD_OPTIONAL },
-   { "/var/opt/cray/alps/spool", "/var/opt/cray/alps/spool", BD_OPTIONAL },
    { 0 }
 };
 
+/* Cray bind-mounts. */
+struct bind BINDS_CRAY[] = {
+   { "/var/lib/hugetlbfs",       "/var/lib/hugetlbfs",       BD_OPTIONAL },
+};
+
+/* Cray Gemini/Aries interconnect bind-mounts. */
+struct bind BINDS_CRAY_UGNI[] = {
+   { "/etc/opt/cray/wlm_detect", "/etc/opt/cray/wlm_detect", BD_OPTIONAL },
+   { "/opt/cray/wlm_detect",     "/opt/cray/wlm_detect",     BD_OPTIONAL },
+   { "/opt/cray/alps",           "/opt/cray/alps",           BD_OPTIONAL },
+   { "/opt/cray/udreg",          "/opt/cray/udreg",          BD_OPTIONAL },
+   { "/opt/cray/ugni",           "/opt/cray/ugni",           BD_OPTIONAL },
+   { "/opt/cray/xpmem",          "/opt/cray/xpmem",          BD_OPTIONAL },
+   { "/var/opt/cray/alps",       "/var/opt/cray/alps",       BD_OPTIONAL },
+};
+
+/* Known Cray Shasta bind-mounts. */
+struct bind BINDS_CRAY_SHASTA[] = {
+   { "/var/spool/slurmd",        "/var/spool/slurmd",        BD_OPTIONAL },
+};
```

Comment:
Is there a reason these aren't just more entries in `BINDS_DEFAULT`? The treatment looks the same.

Reply:
Good question.

The binds to enable Cray interconnect access differs between Aries and Slingshot. The original idea behind structuring them this way was: 1) for clarity: make clear distinctions between shared, aries-specific, and slingshot-specific convenience binds; and 2) allow for explicit bind arguments, e.g., `--bind-aries`, and `--bind-slingshot`, in preparation for Cray PE containers.

Recall that Cray PE containers are going to do cray things. Software will be installed to `/opt` and the various default binds above. It seems like such a container will likely have issues running on different flavors of Cray when all these optional binds are picked up. The thought then would be to make `BINDS_CRAY_UGNI` and `BIND_CRAY_SHASTA` binds non-default, and correspond to the host interconnect you wish to access, e.g., aries or slingshot.

For now, all this can probably go in a comment and all these paths added to default binds. When we have Cray PE containers we could revisit.